src/webhooks/handleSubscriptionUpdated.ts

```diff
@@ -20,20 +22,38 @@ export default async function handleSubscriptionUpdated(
   if (lifetime) {
     return;
   }
-  const isSubscriptionCanceled = subscription.status === 'canceled';
 
-  const bytesSpace = isSubscriptionCanceled
-    ? FREE_PLAN_BYTES_SPACE
-    : parseInt((subscription.items.data[0].price.metadata as unknown as PriceMetadata).maxSpaceBytes);
-
-  const planId = isSubscriptionCanceled ? FREE_INDIVIDUAL_TIER : (subscription.items.data[0].price.product as string);
+  const productId = subscription.items.data[0].price.product as string;
+  const { metadata : productMetadata } = await paymentService.getProduct(productId);
+  const productType = productMetadata?.type === UserType.Business ? UserType.Business : UserType.Individual;
 
   try {
-    await cacheService.clearSubscription(customerId);
+    await cacheService.clearSubscription(customerId, productType);
   } catch (err) {
     log.error(`Error in handleSubscriptionUpdated after trying to clear ${customerId} subscription`);
   }
 
+  if (productType === UserType.Business) {
+    const customer = await paymentService.getCustomer(customerId);
+    if (customer.deleted) {
+      log.error(
+        `Customer object could not be retrieved in subscription updated handler with id ${customer.id}`,
+      );
+      return;
+    }
+    const { maxSpaceBytes: priceMaxSpaceBytes } = subscription.items.data[0].price.metadata as PriceMetadata;
+    const amountOfSeats = subscription.items.data[0]!.quantity!;
+
```

Comment:
What happens if the subscription is canceled in the Business plan case? It does not seem to be handled right now